checks.d/sidekiq.py

```diff
@@ -0,0 +1,246 @@
+    def _collect_metadata(self, info):
+        if info and 'redis_version' in info:
+            self.service_metadata('version', info['redis_version'])
+
+    # Taken from redisdb.py, retrieves a cached redis connection.
+    def _get_conn(self, instance):
+        key = self._generate_instance_key(instance)
+        if key not in self.connections:
+            try:
+
+                # Only send useful parameters to the redis client constructor
+                list_params = ['host', 'port', 'db', 'password', 'socket_timeout',
+                               'connection_pool', 'charset', 'errors', 'unix_socket_path']
+
+                # Set a default timeout (in seconds) if no timeout is specified in the instance config
+                instance['socket_timeout'] = instance.get('socket_timeout', 5)
+
+                connection_params = dict((k, instance[k]) for k in list_params if k in instance)
+
+                self.connections[key] = redis.Redis(**connection_params)
+
+            except TypeError:
+                raise Exception("You need a redis library that supports authenticated connections. Try sudo easy_install redis.")
+
+        return self.connections[key]
+
+    def _check_sidekiq(self, instance, custom_tags):
+        conn = self._get_conn(instance)
+
+        tags, tags_to_add = self._get_tags(custom_tags, instance)
+
+        # Ping the database for info, and track the latency.
+        # Process the service check: the check passes if we can connect to Redis
+        start = time.time()
+        info = None
+        try:
+            info = conn.info()
+            status = AgentCheck.OK
+            self.service_check('sidekiq.redis.can_connect', status, tags=tags_to_add)
+            self._collect_metadata(info)
+        except ValueError, e:
+            status = AgentCheck.CRITICAL
+            self.service_check('sidekiq.redis.can_connect', status, tags=tags_to_add)
+            raise
+        except Exception, e:
+            status = AgentCheck.CRITICAL
+            self.service_check('sidekiq.redis.can_connect', status, tags=tags_to_add)
+            raise
+
+        latency_ms = round((time.time() - start) * 1000, 2)
+        self.gauge('sidekiq.redis.info.latency_ms', latency_ms, tags=tags)
+
+        stats = Stats(conn, instance.get('namespace'))
+
+        # Check global sidekiq stats
+        self.gauge('sidekiq.processed', stats.processed, tags=tags)
+        self.gauge('sidekiq.failed', stats.failed, tags=tags)
+        self.gauge('sidekiq.scheduled_size', stats.scheduled_size, tags=tags)
+        self.gauge('sidekiq.retry_size', stats.retry_size, tags=tags)
+        self.gauge('sidekiq.dead_size', stats.dead_size, tags=tags)
+        self.gauge('sidekiq.processes_size', stats.processes_size, tags=tags)
+        self.gauge('sidekiq.workers_size', stats.workers_size, tags=tags)
+
+        # Check queue stats
+        for queue, stat in stats.queue_stats.iteritems():
```

Comment:
@bmarini The Couch integration has an example of a whitelist that might be a good starting point.
https://github.com/DataDog/dd-agent/pull/2066#discussion_r47966153

Theres an example of a hard limit there too.